obj: fix helgrind issues with pmalloc

pbalcer opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/nvml/1664)
<!-- Reviewable:end -->

wojtuss commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1664#-:-KccNj9zTeJxNjWNK_o4:bnfp4nl)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/common/valgrind_internal.h, line 62 at r1](https://reviewable.io:443/reviews/pmem/nvml/1664#-Kccf192mwdDTKMdGdkh:-Kccf192mwdDTKMdGdki:b-ocetrz) ([raw file](https://github.com/pmem/nvml/blob/638e3aa2c076f736a6b7845d7478ce7ab3a7f1f4/src/common/valgrind_internal.h#L62)):*
> ```objc
> #if defined(USE_VG_HELGRIND)
> #include <valgrind/helgrind.h>
> #elif defined(USE_VG_DRD)
> ```

elif? can't both be defined?

---

*[src/test/obj_heap/obj_heap.c, line 269 at r1](https://reviewable.io:443/reviews/pmem/nvml/1664#-KccfuzFE1yEwhuZkJp5:-KccfuzGc1nqfrE2AwO6:b9j992t) ([raw file](https://github.com/pmem/nvml/blob/638e3aa2c076f736a6b7845d7478ce7ab3a7f1f4/src/test/obj_heap/obj_heap.c#L269)):*
> ```C
> 
> 	/* the old block should be unclaimed now */
> 	UT_ASSERTeq(old_run.m_ops->claim(&old_run), 0);
> ```

hmmm, doesn't it change the behavior of debug version?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1664)*
<!-- Sent from Reviewable.io -->

pbalcer commented:
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/common/valgrind_internal.h, line 62 at r1](https://reviewable.io:443/reviews/pmem/nvml/1664#-Kccf192mwdDTKMdGdkh:-KcvzfujzngCCCHJNBox:b-to14sw) ([raw file](https://github.com/pmem/nvml/blob/638e3aa2c076f736a6b7845d7478ce7ab3a7f1f4/src/common/valgrind_internal.h#L62)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

elif? can't both be defined?
</blockquote></details>

Both of these files contain, at first glance, different definitions of the same macros, which confused me. But as it turns out, drd checks if helgrind.h is already included and doesn't define its macros AND those macros do exactly the same thing, but you have to get one level deeper :)

Fixed.

---

*[src/test/obj_heap/obj_heap.c, line 269 at r1](https://reviewable.io:443/reviews/pmem/nvml/1664#-KccfuzFE1yEwhuZkJp5:-Kcw-JxDm1sm3c82zzO6:bbki7v7) ([raw file](https://github.com/pmem/nvml/blob/638e3aa2c076f736a6b7845d7478ce7ab3a7f1f4/src/test/obj_heap/obj_heap.c#L269)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

hmmm, doesn't it change the behavior of debug version?
</blockquote></details>

It's a test, so it doesn't matter. I removed the old function because it wasn't used anywhere and was broken anyway.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1664)*
<!-- Sent from Reviewable.io -->

pbalcer commented:
Can we please get this merged? The change is simple enough and currently the long tests are broken.

krzycz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1664#-:-Kcw88DsH16H1TlYc6uE:bnfp4nl)*
<!-- Sent from Reviewable.io -->
